### DISCUSSION TITLE
"Next" button for pagination appears when there is no next page / "Last day" results don't fetch the passed 24hrs?

### DISCUSSION THREAD

**Discussion by mdillondc** [searxng/searxng#3538] [Q&A] [answered]
Version: `2024.5.31+18fb701be`

1. "Next" button (pagination) is not hidden when there are no more pages in pagination.

Example: Search "Ukraine war" in section "News" and set timeframe to "Last day". Sometimes there aren't all that many results, and there is no "Next" page of results to navigate to. Yet, the "Next" button appears anyway and clicking it simply refreshes the same results. Shouldn't "Next" be hidden if there is no "Next" page?

2. The "Last day" timeframe for searches behaves in a weird way.

It might be by design, but I am curious regardless. When I select "Last day" I originally thought that meant "the past 24 hours from when the search is performed", but based on what I'm seeing it looks like "Last day" is not looking 24 hours into the past, but rather starts at midnight (or something similar)?

Thank you for any clarification on these points. Maybe I am missing something?

All the best!

**Comment by unixfox** [comment]
Because we don't know if there is going to be a next page. Not all services tell us if there is going to have a next page so we let the user try.

If there was no next page then we display an error message.

**Reply by mdillondc** [reply]
That makes sense. I appreciate your understanding and explanation.

**Comment by mdillondc** [comment]
What about the "Last day" timeframe?

**Reply by unixfox** [reply, marked as answer]
Again it depends on the underlying engine. It's the engine that provides this info.
We are only relaying it.
